pyk/src/pyk/cterm/cterm.py

```diff
@@ -253,6 +253,26 @@ def remove_useless_constraints(self, keep_vars: Iterable[str] = ()) -> CTerm:
         return CTerm(self.config, new_constraints)
 
 
+def cterm_match(cterm1: CTerm, cterm2: CTerm) -> CSubst | None:
+    """Find a substitution which can instantiate `cterm1` to `cterm2`.
+
+    Args:
+        cterm1: `CTerm` to instantiate to `cterm2`.
+        cterm2: `CTerm` to instantiate `cterm1` to.
+
+    Returns:
+        A `CSubst` which can instantiate `cterm1` to `cterm2`, or `None` if no such `CSubst` exists.
+    """
+    subst = cterm1.config.match(cterm2.config)
+    if subst is None:
+        return None
+    source_constraints = [subst(c) for c in cterm1.constraints]
+    constraints = [c for c in cterm2.constraints if c not in source_constraints] + [
+        c for c in source_constraints if c not in cterm2.constraints
+    ]
```

Comment:
I think the second line here (`c for c in source_constraints if c not in cterm2.constraints`) is probably not needed. It won't add any constraints to `cterm1` when the csubst is applied that are new.

Reply:
Yes, deleted this line!